crates/core/tedge_agent/src/file_transfer_server/request_files.rs

```diff
@@ -0,0 +1,84 @@
+use std::fmt;
+use std::ops::Deref;
+use std::sync::Arc;
+
+use axum::extract::FromRequestParts;
+use axum::extract::Path;
+use axum::http::request::Parts;
+use camino::Utf8Path;
+use camino::Utf8PathBuf;
+
+use super::error::FileTransferRequestError;
+use super::http_rest::HttpConfig;
+
+/// The paths inferred from a request to the File Transfer Service
+pub struct FileTransferPaths {
```

Comment:
```suggestion
pub struct FileTransferPath {
```
Nitpick(Optional): Although this struct contains two paths, they still point to the same logical entity, right? So, a plural name just sounds weird.